calculator_test.go

```diff
@@ -173,19 +356,194 @@ func TestDivide(t *testing.T) {
 		},
 	}
 
-	for _, test := range successTestCases {
-		got, err := calculator.Divide(test.a, test.b)
+	for _, tc := range tcs {
+		got, err := calculator.Divide(tc.a, tc.b)
+		errReceived := err != nil
+		if tc.errExpected != errReceived {
+			t.Errorf("Test Divide (%f, %f) (%s) failed: Unexpected error received",
+				tc.a, tc.b, tc.name)
+		} else {
+			if !tc.errExpected && !almostEqual(tc.want, got) {
+				t.Errorf("Test Divide (%f, %f) (%s) failed: want %f, got %f",
+					tc.a, tc.b, tc.name, tc.want, got)
+			}
+		}
+	}
+}
+
+func TestDivideN(t *testing.T) {
+	t.Parallel()
+	var tcs = []struct {
+		name        string
+		want        float64
+		nums        []float64
+		errExpected bool
+	}{
+		{
+			name:        "Two positive numbers",
+			want:        5,
+			nums:        []float64{10, 2},
+			errExpected: false,
+		},
+		{
+			name:        "Single number in slice ",
+			want:        5,
+			nums:        []float64{5},
+			errExpected: false,
+		},
+		{
+			name:        "Empty Slice ",
+			want:        0,
+			nums:        []float64{},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 numbers",
+			want:        1,
+			nums:        []float64{20, 5, 2, 2},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions",
+			want:        2,
+			nums:        []float64{5.5, 1.1, 2.5},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions starting with 0",
+			want:        0,
+			nums:        []float64{0, 1.1, 2.5},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions",
```

Comment:
Actually, the name of this test case should be something like 'Divide by zero', shouldn't it?